pkg/inject/sidecar_builder.go

```diff
@@ -190,6 +193,16 @@ func buildEnvoySidecar(vars EnvoyTemplateVariables, env map[string]string) (core
 		},
 	}
 
+	if vars.WaitUntilProxyReady {
+		envoy.Lifecycle.PostStart = &corev1.Handler{
+			Exec: &corev1.ExecAction{Command: []string{
+				"sh", "-c", fmt.Sprintf("if [ -f /tmp/agent.sock ]; then APPNET_AGENT_POLL_ENVOY_READINESS_TIMEOUT_S=%d "+
```

Comment:
add comment about why we need to check this /tmp/agent.sock file

Reply:
explore other avenues for checking envoy/agent version since agent socket file could not exist when postStart is run

Reply:
add an `else` echoing version of envoy is not supported with postStart